### ISSUE TITLE
cuda_runtime_api.h: No such file or directory

### ISSUE THREAD

**Issue by kishorekuttalamr** [marcoslucianops/DeepStream-Yolo#639]
![Image](https://github.com/user-attachments/assets/46ffc065-74b2-42f4-9089-b6629c09242f)

(base) friday@friday-ASUS-TUF-Gaming-F15-FX507VI-FX507VI:~/Documents/gitYOLO/DeepStream-Yolo$ make -C nvdsinfer_custom_impl_Yolo clean && make -C nvdsinfer_custom_impl_Yolo
make: Entering directory '/home/friday/Documents/gitYOLO/DeepStream-Yolo/nvdsinfer_custom_impl_Yolo'
rm -rf libnvdsinfer_custom_impl_Yolo.so
rm -rf nvdsinfer_yolo_engine.o nvdsparsebbox_Yolo.o utils.o yolo.o yoloPlugins.o layers/activation_layer.o layers/batchnorm_layer.o layers/channels_layer.o layers/convolutional_layer.o layers/deconvolutional_layer.o layers/implicit_layer.o layers/pooling_layer.o layers/reorg_layer.o layers/route_layer.o layers/sam_layer.o layers/shortcut_layer.o layers/slice_layer.o layers/upsample_layer.o nvdsparsebbox_Yolo_cuda.o yoloForward.o yoloForward_nc.o yoloForward_v2.o
make: Leaving directory '/home/friday/Documents/gitYOLO/DeepStream-Yolo/nvdsinfer_custom_impl_Yolo'
make: Entering directory '/home/friday/Documents/gitYOLO/DeepStream-Yolo/nvdsinfer_custom_impl_Yolo'
g++ -c  -o nvdsinfer_yolo_engine.o -Wall -std=c++11 -shared -fPIC -Wno-error=deprecated-declarations -I/opt/nvidia/deepstream/deepstream/sources/includes -I/usr/local/cuda-12.6/include nvdsinfer_yolo_engine.cpp
In file included from /usr/include/x86_64-linux-gnu/NvInferLegacyDims.h:22,
                 from /usr/include/x86_64-linux-gnu/NvInfer.h:21,
                 from /opt/nvidia/deepstream/deepstream/sources/includes/nvdsinfer_custom_impl.h:117,
                 from nvdsinfer_yolo_engine.cpp:28:
/usr/include/x86_64-linux-gnu/NvInferRuntimeBase.h:24:10: fatal error: cuda_runtime_api.h: No such file or directory
   24 | #include <cuda_runtime_api.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:86: nvdsinfer_yolo_engine.o] Error 1
make: Leaving directory '/home/friday/Documents/gitYOLO/DeepStream-Yolo/nvdsinfer_custom_impl_Yolo'

Using dGPU RTX4070. 
DeepStream Version: 7.1.0
TensorRT Version:10.10.0.31-1
NVIDIA GPU Driver Version:  570.133.07(CUDA version: 12.6)

**Comment by jmudarra23** [comment]
If you are using DeepStream 7.1 you will probably have to modify several paths in the Makefile. On lines 45 and 57, change

`-I/opt/nvidia/deepstream/deepstream/sources/includes`

to

`-I/opt/nvidia/deepstream/deepstream/deepstream-7.1/sources/includes`

Try compiling again and it should work.

**Comment by marcoslucianops** [comment]
Do you have CUDA 12.6 installed?

**Comment by kishorekuttalamr** [comment]
I got the code working, but I noticed a significant drop in performance. When running the model with standard PyTorch, I average around 150 FPS. However, using the DeepStream app, I only get about 36 FPS — which is roughly 5 times slower.

Is this drop in FPS a common issue when switching to DeepStream? If so, what typically causes this slowdown? Or could I be missing something in the config.txt file that's affecting performance?

**Comment by marcoslucianops** [comment]
You can set FP16 mode (`network-mode=2`. Also set `[sink0] sync=0`.
